### PR TITLE
Macros: Support $_interval[_ms] variable

### PR DESCRIPTION
**Pull request by dprokop**
Closes https://github.com/grafana/scenes/issues/486

An approach to support `__interval[_ms]` macros. 

I've also tried a different approach which would resolve the inteval on the SceneQueryRunner without the need for storing `_lastRequest`, but it has become super tricky as this function would have to by async (in oroder to get interval from the data source) and this would require us to also have async `FormatVariable` :(

But with this `SceneDataProvide` API addition, we can easily implement interval retrieval for SharedDataQueryProvider - see PR in core: https://github.com/grafana/grafana/pull/78952

@torkelo  any thoughts on the default 1s interval returned?

<details>
  <summary>📦 Published PR as canary version: <code>1.25.0--canary.487.7060253941.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/scenes@1.25.0--canary.487.7060253941.0
  # or 
  yarn add @grafana/scenes@1.25.0--canary.487.7060253941.0
  ```
</details>

### CODE REVIEW

File: packages/scenes/src/querying/SceneQueryRunner.ts [hunk 1]
```diff
@@ -476,7 +481,7 @@ export class SceneQueryRunner extends SceneObjectBase<QueryRunnerState> implemen
     return [request, secondaryRequest];
   };
 
-  private onDataReceived = (data: PanelData) => {
+  private onDataReceived = (data: PanelData, request: [DataQueryRequest, DataQueryRequest | undefined]) => {
```
Comment (torkelo) [inline packages/scenes/src/querying/SceneQueryRunner.ts:484]: PanelData already has the request I think

File: packages/scenes/src/querying/SceneQueryRunner.ts [hunk 2]
```diff
@@ -560,6 +565,13 @@ export class SceneQueryRunner extends SceneObjectBase<QueryRunnerState> implemen
     this._adhocFilterSet = set;
     this._adhocFilterSub = this._adhocFilterSet?.subscribeToState(() => this.runQueries());
   }
+
+  public getInterval(): IntervalValues {
+    return {
+      interval: this.state._lastRequest?.[0].interval ?? '1s',
```
Comment (torkelo) [inline packages/scenes/src/querying/SceneQueryRunner.ts:571]: I think we can just do this.state.data.request (but on mobile so not sure)

**Review by torkelo — APPROVED** [review]
Nice idea with using a macro! Looks great.

But not sure we need the changes to scene query runner, PanelData should have the request and the interval, but not 100% sure.

File: packages/scenes/src/querying/SceneDataTransformer.ts [hunk 3]
```diff
@@ -100,6 +100,10 @@ export class SceneDataTransformer extends SceneObjectBase<SceneDataTransformerSt
     return this._results;
   }
 
+  public getInterval() {
+    return this.getSourceData().getInterval?.() || { interval: '1s', intervalMs: 1000 };
```
Comment (torkelo) [inline packages/scenes/src/querying/SceneDataTransformer.ts:104]: Ah, do we preserve PanelData.request after transform, I think that should be preserved

Comment (dprokop) [reply packages/scenes/src/querying/SceneDataTransformer.ts:104]: yup, it's preserved

File: packages/scenes/src/core/types.ts [hunk 4]
```diff
@@ -202,6 +203,7 @@ export interface SceneDataProvider extends SceneObject<SceneDataState> {
   isDataReadyToDisplay?: () => boolean;
   cancelQuery?: () => void;
   getResultsStream?(): Observable<SceneDataProviderResult>;
+  getInterval?(): IntervalValues;
```
Comment (torkelo) [inline packages/scenes/src/core/types.ts:206]: Do we really need this? Can't we just read it from data.request?

**Pull request merged**